### PR TITLE
[Merged by Bors] - feat: add `measurableSet_tendsto_nhds`

### PR DESCRIPTION
**Pull request by RemyDegenne**
Add `measurableSet_tendsto`: the set of points for which a measurable sequence of functions converges to a given filter is measurable.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Constructions/Polish.lean [hunk 1]
```diff
@@ -921,6 +921,30 @@ theorem isClopenable_iff_measurableSet
...
+    Set.setOf_forall]
+  refine MeasurableSet.biInter Set.countable_univ fun K _ ↦ ?_
+  simp_rw [Set.setOf_exists, true_and]
+  refine MeasurableSet.iUnion fun N ↦ ?_
+  simp_rw [image_subset_iff]
+  change MeasurableSet {x | ∀ i ∈ u N, i ∈ (fun n ↦ f n x) ⁻¹' Metric.ball c (1 / (K + 1))}
+  simp_rw [Set.setOf_forall]
+  refine MeasurableSet.biInter (to_countable (u N)) fun i _ ↦ ?_
+  simp only [one_div, mem_preimage, Metric.mem_ball]
+  exact measurableSet_lt (Measurable.dist (hf i) measurable_const) measurable_const
```
[the remark this answers is not in the document]
Comment (ADedecker) [reply Mathlib/MeasureTheory/Constructions/Polish.lean:946]: Here's an even better version based on the discussion on Zulip, which allows more general filters on the target!
```suggestion
lemma measurableSet_tendsto [MeasurableSpace γ] [Countable ι] {l : Filter ι} (l' : Filter γ)
    [l.IsCountablyGenerated] [l'.IsCountablyGenerated] [hl' : l'.IsMeasurablyGenerated]
    {f : ι → β → γ} (hf : ∀ i, Measurable (f i)) :
    MeasurableSet { x | Tendsto (fun n ↦ f n x) l l' } := by
  rcases l.exists_antitone_basis with ⟨u, hu⟩
  rcases (Filter.hasBasis_self.mpr hl'.exists_measurable_subset).exists_antitone_subbasis with
    ⟨v, v_meas, hv⟩
  simp only [hu.tendsto_iff hv.toHasBasis, true_imp_iff, true_and, setOf_forall, setOf_exists]
  exact .iInter fun n ↦ .iUnion fun _ ↦ .biInter (to_countable _) fun i _ ↦
    (v_meas n).2.preimage (hf i)
```

[the remark this answers is not in the document]
Comment (RemyDegenne) [reply Mathlib/MeasureTheory/Constructions/Polish.lean]: Thanks for the suggestion! I've replaced the lemma with your version.
I have added another related result that I will also need, about convergence to a given function. I still need to move the two lemmas to an appropriate place.

**Comment by RemyDegenne** [comment]
I moved `measurableSet_tendsto` to an earlier file (right after the definition of `IsMeasurablyGenerated`). According to `#find_home!`, the second one should stay in the Polish file (we don't have a measure theory file about metrizable spaces).

**Comment by ADedecker** [comment]
Thanks!

bors merge

**Pull request closed without merging**